src/python/pants/backend/docker/util_rules/docker_build_context.py

```diff
@@ -289,9 +290,19 @@ async def create_docker_build_context(request: DockerBuildContextRequest) -> Doc
     )
 
     # Package binary dependencies for build context.
-    embedded_pkgs = await MultiGet(
-        Get(BuiltPackage, PackageFieldSet, field_set)
+    environment_names_per_field_set = await MultiGet(
```

Comment:
Noting that this is happening over in `package.py` too, it looks like this pattern of requesting a set of objects corresponding to different environments is likely to be a frequent one -- and one that took me a bit of reading to wrap my head around.

Any chance of factoring that out into something reusable?

Reply:
Yea, that's probably worth doing. The three use-cases of environments-on-roots so far all use `FieldSetsPerTargetRequest` ... will see if it can be applied there generically.